internal/static/integrations/ssh_exporter/ssh_client.go

```diff
@@ -0,0 +1,216 @@
+package ssh_exporter
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"time"
+
+	"os/user"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/crypto/ssh/knownhosts"
+)
+
+type SSHClient struct {
+	config  *ssh.ClientConfig
+	host    string
+	port    int
+	logger  log.Logger
+	timeout time.Duration
+}
+
+var sshKeyscanCommand = func(targetAddress string) ([]byte, error) {
+	cmd := exec.Command("ssh-keyscan", "-H", targetAddress)
+	return cmd.Output()
+}
+
+func ensureKnownHosts(knownHostsPath, targetAddress string) error {
+	// On Windows, skip known_hosts handling
+	if runtime.GOOS == "windows" {
+		return nil
+	}
+	// Ensure .ssh directory exists
+	if err := os.MkdirAll(filepath.Dir(knownHostsPath), 0700); err != nil {
+		return fmt.Errorf("failed to create .ssh directory: %w", err)
+	}
+
+	var knownHostsContent []string
+	if _, err := os.Stat(knownHostsPath); err == nil {
+		content, err := os.ReadFile(knownHostsPath)
+		if err != nil {
+			return fmt.Errorf("failed to read known_hosts file: %w", err)
+		}
+		knownHostsContent = strings.Split(string(content), "\n")
+	}
+
+	var output []byte
+	var scanErr error
+	// Keep retrying until we successfully fetch the host key
+	for {
+		output, scanErr = sshKeyscanCommand(targetAddress)
+		if scanErr == nil && len(output) > 0 {
+			break
+		}
+		fmt.Printf("failed to fetch host key for %s: %v; retrying in 1s...\n", targetAddress, scanErr)
+		time.Sleep(time.Second)
+	}
```

Comment:
In the test that you mentioned is failing - this loop is not terminating before test timeout.